students/PhongT/lesson06/mailroom.py

```diff
@@ -0,0 +1,176 @@
+    """
+    Find whether or not the input donor is in the db.
+    :param donor_name: the donor to search for in donor db
+    :return: donor key id if there is existing one in db. Otherwise return None
+    """
+    for index, existing_donor in enumerate(donors_data):
+        if existing_donor["name"].lower().strip() == donor_name.lower().strip():
+            return index
+    return None
+
+
+def add_donor_to_db(donor_name, amount, record_id=None):
+    """
+    Add a donor to the db.
+    :param donor_name: a donor to add
+    :param amount: donation amount
+    :param record_id: record id of an existing entry in db. Equals None if is new record
+    :return: a new added donor entry in db
+    """
+    if record_id is None:
+        logger.info('Add a new record entry')
+        donors_data.append({"name": donor_name, "donations": [amount]})
+        return donors_data[-1]
+    else:
+        logger.info('Update an existing record entry')
+        donors_data[record_id]["donations"].append(amount)
+        return donors_data[record_id]
+
+
+def get_thankyou_message(donor):
+    """ generate a formatted thank you letter message"""
+    message = '''Dear {}, 
+            Thank you so much for your generosity with your most recent donation of ${}. 
+            It will be put to very good use.
+            Sincerely.'''
+    return message.format(donor["name"], donor["donations"][-1])
+
+
+def print_donor_list():
+    """ print out list of current donors"""
+    print('Below are the existing donors: ')
+    for donor in donors_data:
+        print('\t- ', donor["name"], ' ', donor["donations"])
+
+
+def print_report():
+    """ print a list of your donors, sorted by total historical donation amount. """
+    width = 68
+    dash = "-" * width
+    print(dash)
+    header = ("Donor Name", "Total Given", "Num Gifts", "Average Gift")
+    print("{:20} | {:15} | {:10} | {:12}".format(*header))
+    print(dash)
+    for index, donor in enumerate(donors_data):
+        name = donor["name"]
+        total = sum(donor["donations"])
+        num_gift = len(donor["donations"])
+        average = total/num_gift
+        print("{:22} ${:12,.2f} {:12d}     ${:12,.2f}".format(name, total, num_gift, average ))
+    print(dash)
+
+
+def write_letters_to_file():
+    """
+    goes through all the donors in donor data structure, generates a thank you
+    letter, and writes it to disk as a text file.
+    :return: n/a
+    """
+    for donor in donors_data:
+        message = get_thankyou_message(donor)
+        filename = donor["name"].replace(" ", "_") + ".rpt"
+        logger.info("write letter to:" + donor["name"])
+        with open(filename, 'w') as outfile:
+            outfile.write(message)
+
+
+def exit():
+    """ exit the program """
+    sys.exit(0)
+
+
+def main_menu():
+    menu_dict = {"1": print_thank_you,
+                "2": print_report,
+                "3": write_letters_to_file,
+                "4": exit}
```

Comment:
best to just make this `sys.exit` rather than call another function to call `sys.exit`